backend/src/appointment/database/models.py

```diff
@@ -98,7 +98,17 @@ def get_columns(self) -> list:
     time_updated = Column(DateTime, server_default=func.now(), default=func.now(), onupdate=func.now(), index=True)
 
 
-class Subscriber(Base):
+class HasSoftDelete:
+    """Mixing in a column to support deletion without removing the record"""
+    time_deleted = Column(DateTime, nullable=True)
+
+    @property
+    def is_deleted(self):
+        """A record is marked deleted if a delete time is set."""
+        return self.time_deleted is not None
+
+
+class Subscriber(HasSoftDelete, Base):
```

Comment:
I added this mixin separately. Now we can give all models for which we want soft delete this class.